sn_node_manager/src/add_services/mod.rs

```diff
@@ -0,0 +1,334 @@
+// Copyright (C) 2024 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+pub mod config;
+#[cfg(test)]
+mod tests;
+
+use self::config::{
+    AddDaemonServiceOptions, AddFaucetServiceOptions, AddServiceOptions,
+    InstallFaucetServiceCtxBuilder, InstallNodeServiceCtxBuilder,
+};
+use crate::{config::create_owned_dir, VerbosityLevel, DAEMON_SERVICE_NAME};
+use color_eyre::{eyre::eyre, Help, Result};
+use colored::Colorize;
+use service_manager::ServiceInstallCtx;
+use sn_service_management::{
+    control::ServiceControl, DaemonServiceData, FaucetServiceData, NodeRegistry, NodeServiceData,
+    ServiceStatus,
+};
+use std::{
+    ffi::OsString,
+    net::{IpAddr, Ipv4Addr, SocketAddr},
+};
+
+/// Install safenode as a service.
+///
+/// This only defines the service; it does not start it.
+///
+/// There are several arguments that probably seem like they could be handled within the function,
+/// but they enable more controlled unit testing.
+pub async fn add(
+    options: AddServiceOptions,
```

Comment:
Can we make the function and the options explicit here, as we have it for daemon and faucet? `add_node` and `AddNodeServiceOptions`.